src/test/java/com/artipie/rpm/hm/StorageHasMetadata.java

```diff
@@ -0,0 +1,115 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.cactoos.text.FormattedText;
+import org.hamcrest.Matcher;
+import org.hamcrest.core.AllOf;
+import org.llorllale.cactoos.matchers.MatcherOf;
+
+/**
+ * Storage has metadata matcher.
+ * @since 0.11
+ * @todo #311:30min Create proper unit test for this class (use metadata examples from test
+ *  resources), do not forget to test mismatches descriptions.
+ * @todo #311:30min Add one more check (here or create another matcher) for repomd.xml: we need to
+ *  verify that exactly one repomd is present in storage and has info about metadatas, as example
+ *  check RpmITCase#assertion(). Repomd should be verified at least in RpmTest and RpmITCase.
+ *  @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class StorageHasMetadata extends AllOf<Storage> {
+
+    /**
+     * Ctor.
+     * @param expected Amount of expected items in metadata
+     * @param filelists Need filelist
+     * @param temp Temp dir to unpack xml
+     */
+    public StorageHasMetadata(final int expected, final boolean filelists, final Path temp) {
+        super(StorageHasMetadata.matchers(expected, filelists, temp));
+    }
+
+    /**
+     * List of matchers.
+     * @param expected Amount of expected items in metadata
+     * @param filelists Need filelist
+     * @param temp Temp dir to unpack xml
+     * @return Matchers list
+     */
+    private static List<Matcher<? super Storage>> matchers(
+        final int expected, final boolean filelists, final Path temp
+    ) {
+        return new XmlPackage.Stream(filelists).get().map(
+            pckg -> new MatcherOf<Storage>(
+                storage -> hasMetadata(storage, temp, pckg, expected),
+                new FormattedText("Metadata %s has %d rpm packages", pckg.name(), expected)
+            )
+        ).collect(Collectors.toList());
+    }
+
+    /**
+     * Obtains metadata xml from storage and checks that
+     * this metadata has correct amount of packages.
+     * @param storage Storage
+     * @param temp Temp dir to unpack xml
+     * @param pckg Package type
+     * @param expected Amount of expected items in metadata
+     * @return True if metadata is correct
+     * @throws IOException On IO error
+     * @throws InterruptedException On storage op error
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    private static boolean hasMetadata(
+        final Storage storage, final Path temp, final XmlPackage pckg, final int expected
+    ) throws InterruptedException, IOException {
+        final BlockingStorage bsto = new BlockingStorage(storage);
+        final List<Key> repodata = bsto.list(new Key.From("repodata")).stream()
+            .filter(key -> key.string().contains(pckg.filename())).collect(Collectors.toList());
+        if (repodata.size() != 1) {
+            throw new IOException(
```

Comment:
@olenagerasimova should not this case produce `false` result, indicating that metadata in storage is not correct?